src/main/java/org/distributed/systems/chord/actors/NodeActor.java

```diff
@@ -188,26 +229,26 @@ public Receive createReceive() {
                     }
                 })
                 .match(Connected.class, conn -> {
-                    System.out.println("MemCache Client connected");
                     manager.tell(conn, getSelf());
                     ActorRef memcacheHandler = getContext().actorOf(Props.create(MemcachedActor.class, storageActorRef));
```

Comment:
Is this the issue? However, if I change this to `getSelf()` everything goes to the shit. :sweat_smile: 

I would assume it works like this MemCacheActor -> Node -> StorageActor.
Currently it goes like MemCacheActor -> StorageActor...

I am clueless though :(

Reply:
@stienis96 Any idea?

Reply:
@l8518 Yes that was mostly the issue. I can't test it fully now though, my ubuntu vm is a bit limited. Can you run your test one more time with a few nodes?

Reply:
:fire: :dancing_men: